plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -842,6 +842,24 @@ public void testInPredicatePushdown()
         }
     }
 
+    @Test
+    public void testCastPushdown()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_cast_pushdown",
+                "(id bigint, id2 varchar(1))",
+                List.of(
+                        "1, 'b'",
+                        "2, 'c'",
+                        "3, 'c'",
+                        "4, 'd'",
+                        "5, 'f'"))) {
+            assertThat(query("SELECT id FROM " + table.getName() + " WHERE CAST(id AS VARCHAR(1)) = '2' OR id2 = 'd'"))
```

Comment:
Trino and PostgreSQL have different CAST behavior for BIGINT -> varchar(1) cast.

```
trino> SELECT CAST(BIGINT '12' AS varchar(1));
Query 20220509_075440_00000_3ce4u failed: Value 12 cannot be represented as varchar(1)
```

PostgreSQL
```
SELECT CAST(BIGINT '12' AS varchar(1));
1
```

Reply:
Postgres ended up going the Hive way? Does the SQL spec say anything about this?

Reply:
@hashhar see https://github.com/trinodb/trino/issues/552#issuecomment-536233927